src/SqlPersistence/SynchronizedStorage/StorageSessionFeature.cs

```diff
@@ -1,14 +1,23 @@
-﻿using System.Linq;
+﻿using System;
+using System.Linq;
 using NServiceBus;
 using NServiceBus.Features;
 using NServiceBus.ObjectBuilder;
+using NServiceBus.Persistence;
 
 class StorageSessionFeature : Feature
 {
 
     protected override void Setup(FeatureConfigurationContext context)
     {
-        var connectionBuilder = context.Settings.GetConnectionBuilder();
+        var settings = context.Settings;
+        var isSagasEnabled = settings.GetFeatureEnabled<StorageType.Sagas>();
+        var isOutboxEnabled = settings.GetFeatureEnabled<StorageType.Outbox>();
+        if (!isSagasEnabled || !isOutboxEnabled)
```

Comment:
Hoa about a case where somebody enables the persistence explicitly for sagas but not for outbox (because he does not use the outbox)?